ambry-api/src/main/java/com.github.ambry/account/AccountBuilder.java

```diff
@@ -0,0 +1,216 @@
+      short id = container.getId();
+      Container containerToRemove = idToContainerMetadataMap.get(id);
+      if (container.equals(containerToRemove)) {
+        idToContainerMetadataMap.remove(id);
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Sets a list of {@link Container}s for the {@link Account} to build. If any {@link Container} in the list already
+   * exists in the builder, it will be replaced by the one to set.
+   * @param containers A list of {@link Container}s to set.
+   * @return This builder.
+   */
+  public AccountBuilder setContainers(List<Container> containers) {
+    if (containers != null) {
+      for (Container container : containers) {
+        setContainer(container);
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Builds an {@link Account} object. {@code id}, {@code name}, and {@code status} are required before building.
+   * @return An {@link Account} object.
+   * @throws IllegalStateException If any required fields is not set.
+   */
+  public Account build() throws JSONException {
+    checkAccountIdInContainers();
+    checkRequiredFields(version);
+    switch (version) {
+      case ACCOUNT_METADATA_VERSION_1:
+        if (id == null) {
+          throw new IllegalStateException("Cannot build account, id is not set");
+        }
+        if (name == null) {
+          throw new IllegalStateException("Cannot build account, name is not set");
+        }
+        if (status == null) {
+          throw new IllegalStateException("Cannot build account, status is not set");
+        }
+        JSONObject jsonMetadata = new JSONObject();
+        jsonMetadata.put(ACCOUNT_METADATA_VERSION_KEY, version);
+        jsonMetadata.put(ACCOUNT_ID_KEY, id);
+        jsonMetadata.put(ACCOUNT_NAME_KEY, name);
+        jsonMetadata.put(ACCOUNT_STATUS_KEY, status);
+        JSONArray containerArray = new JSONArray();
+        for (Container container : idToContainerMetadataMap.values()) {
+          containerArray.put(container.toJson());
+        }
+        jsonMetadata.put(CONTAINERS_KEY, containerArray);
+        return new Account(jsonMetadata);
+
+      default:
+        throw new IllegalStateException("Unsupported account metadata version=" + version);
+    }
+  }
+
+  /**
+   * Checks if required fields are missing to build.
+   */
+  private void checkAccountIdInContainers() {
+    for (Container container : idToContainerMetadataMap.values()) {
+      if (container.getParentAccountId() != id) {
+        throw new IllegalStateException(
+            "Container does not belong to this Account because containerId=" + container.getParentAccountId()
+                + " is not the same as accountId=" + id);
+      }
+    }
+  }
+
+  /**
+   * Checks required fields to build a {@link Account}.
+   * @param version The version to build.
+   */
+  private void checkRequiredFields(short version) {
+    switch (version) {
+      case CONTAINER_METADATA_VERSION_1:
+        if (id == null) {
```

Comment:
I see that the constructor do take in all 3 args. Shouldn't we do this validation during construction itself and fail for any such invalid values.